src/avram/save_operation.cr

```diff
@@ -210,7 +211,9 @@ abstract class Avram::SaveOperation(T)
 
   # Runs required validation,
   # then returns `true` if all attributes are valid.
+  # Set `valid` to false to force invalid state.
   def valid? : Bool
+    return false unless valid
```

Comment:
don't even waste time running required validations since those may run some queries (i.e. uniqs and such). We are saying it's not valid so the others don't really matter.